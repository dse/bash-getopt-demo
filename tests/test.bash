#!/usr/bin/env bash

exec 2>&1

# control
bin/bash-getopt-demo

# --help
bin/bash-getopt-demo --help

# -- by itself
bin/bash-getopt-demo --verbose -- --verbose one two three

# short options
bin/bash-getopt-demo -1234
bin/bash-getopt-demo -1 -2 -3 4

# long options
bin/bash-getopt-demo --one --two --two=TWO --three=THREE

# invalid cases
bin/bash-getopt-demo -123Z
bin/bash-getopt-demo --one --zed --two

# empty strings; strings containing quotes and/or spaces
bin/bash-getopt-demo "" one "'two'" '"three"' "four five" "'six seven'" '"eight nine"'

# every printable non-alphanumeric ASCII character
bin/bash-getopt-demo ' ' '!' '"' '#' '$' '%' '&' "'" '(' ')' '*' '+' ',' '-' '.' '/'
bin/bash-getopt-demo ':' ';' '<' '=' '>' '?' '@' '[' "\\" ']' '^' '_' '`' '{' '|' '}' '~'

# control characters
bin/bash-getopt-demo $'\a' $'\b' $'\e' $'\E' $'\f' $'\n' $'\r' $'\t' $'\v'
bin/bash-getopt-demo $'\c@' $'\cA' $'\cB' $'\cC' $'\cD' $'\cE' $'\cF' $'\cG'
bin/bash-getopt-demo $'\cH' $'\cI' $'\cJ' $'\cK' $'\cL' $'\cM' $'\cN' $'\cO'
bin/bash-getopt-demo $'\cP' $'\cQ' $'\cR' $'\cS' $'\cT' $'\cU' $'\cV' $'\cW'
bin/bash-getopt-demo $'\cX' $'\cY' $'\cZ' $'\c[' $'\034' $'\c]' $'\c^' $'\c_'

# non-ASCII characters: high-bits and unicode
bin/bash-getopt-demo $'\240' $'\xa0'

