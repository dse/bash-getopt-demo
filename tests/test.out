---
This is the program:
  bin/bash-getopt-demo
invoked with no arguments
verbosity level is 0
one=
two=
three=
no option arguments
no remaining positional arguments
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  --help
Parsed option: --help
usage: bin/bash-getopt-demo [OPTION ...] [ARGUMENT ...]
options:
  -h, --help
  -v, --verbose
  -1, --one
  -2, --two[=value]  (short form -2 takes no argument)
  -3, --three=value  (short form -3 requires an argument)
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  --verbose -- --verbose one two three
Parsed option: --verbose
verbosity level is 1
one=
two=
three=
1 option arguments:
   1. --verbose
4 remaining positional arguments:
   1. --verbose
   2. one
   3. two
   4. three
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  -1234
Parsed option: -1
Parsed option: -2 ()
Parsed option: -3 4
verbosity level is 0
one=one
two=two
three=4
4 option arguments:
   1. -1
   2. -2
   3. -3
   4. 4
no remaining positional arguments
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  -1 -2 -3 4
Parsed option: -1
Parsed option: -2 ()
Parsed option: -3 4
verbosity level is 0
one=one
two=two
three=4
4 option arguments:
   1. -1
   2. -2
   3. -3
   4. 4
no remaining positional arguments
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  --one --two --two=TWO --three=THREE
Parsed option: --one
Parsed option: --two
Parsed option: --two=TWO
Parsed option: --three=THREE 
verbosity level is 0
one=one
two=TWO
three=THREE
4 option arguments:
   1. --one
   2. --two
   3. --two=TWO
   4. --three=THREE
no remaining positional arguments
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  -123Z
Parsed option: -1
Parsed option: -2 ()
Parsed option: -3 Z
verbosity level is 0
one=one
two=two
three=Z
4 option arguments:
   1. -1
   2. -2
   3. -3
   4. Z
no remaining positional arguments
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  --one --zed --two
Parsed option: --one
bin/bash-getopt-demo: unrecognized option '--zed'
Type 'bin/bash-getopt-demo --help' for help.
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  '' one \'two\' \"three\" four\ five \'six\ seven\' \"eight\ nine\"
verbosity level is 0
one=
two=
three=
no option arguments
7 remaining positional arguments:
   1. ''
   2. one
   3. \'two\'
   4. \"three\"
   5. four\ five
   6. \'six\ seven\'
   7. \"eight\ nine\"
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  \  \! \" \# \$ % \& \' \( \) \* + \, - . /
verbosity level is 0
one=
two=
three=
no option arguments
16 remaining positional arguments:
   1. \ 
   2. \!
   3. \"
   4. \#
   5. \$
   6. %
   7. \&
   8. \'
   9. \(
  10. \)
  11. \*
  12. +
  13. \,
  14. -
  15. .
  16. /
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  : \; \< = \> \? @ \[ \\ \] \^ _ \` \{ \| \} \~
verbosity level is 0
one=
two=
three=
no option arguments
17 remaining positional arguments:
   1. :
   2. \;
   3. \<
   4. =
   5. \>
   6. \?
   7. @
   8. \[
   9. \\
  10. \]
  11. \^
  12. _
  13. \`
  14. \{
  15. \|
  16. \}
  17. \~
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  $'\a' $'\b' $'\E' $'\E' $'\f' $'\n' $'\r' $'\t' $'\v'
verbosity level is 0
one=
two=
three=
no option arguments
9 remaining positional arguments:
   1. $'\a'
   2. $'\b'
   3. $'\E'
   4. $'\E'
   5. $'\f'
   6. $'\n'
   7. $'\r'
   8. $'\t'
   9. $'\v'
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  '' $'\001' $'\002' $'\003' $'\004' $'\005' $'\006' $'\a'
verbosity level is 0
one=
two=
three=
no option arguments
8 remaining positional arguments:
   1. ''
   2. $'\001'
   3. $'\002'
   4. $'\003'
   5. $'\004'
   6. $'\005'
   7. $'\006'
   8. $'\a'
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  $'\b' $'\t' $'\n' $'\v' $'\f' $'\r' $'\016' $'\017'
verbosity level is 0
one=
two=
three=
no option arguments
8 remaining positional arguments:
   1. $'\b'
   2. $'\t'
   3. $'\n'
   4. $'\v'
   5. $'\f'
   6. $'\r'
   7. $'\016'
   8. $'\017'
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  $'\020' $'\021' $'\022' $'\023' $'\024' $'\025' $'\026' $'\027'
verbosity level is 0
one=
two=
three=
no option arguments
8 remaining positional arguments:
   1. $'\020'
   2. $'\021'
   3. $'\022'
   4. $'\023'
   5. $'\024'
   6. $'\025'
   7. $'\026'
   8. $'\027'
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  $'\030' $'\031' $'\032' $'\E' $'\034' $'\035' $'\036' $'\037'
verbosity level is 0
one=
two=
three=
no option arguments
8 remaining positional arguments:
   1. $'\030'
   2. $'\031'
   3. $'\032'
   4. $'\E'
   5. $'\034'
   6. $'\035'
   7. $'\036'
   8. $'\037'
---
This is the program:
  bin/bash-getopt-demo
invoked with the following arguments:
  $'\240' $'\240'
verbosity level is 0
one=
two=
three=
no option arguments
2 remaining positional arguments:
   1. $'\240'
   2. $'\240'
